analysers/analyser_osmosis_railway_angle.py

```diff
@@ -0,0 +1,111 @@
+## the Free Software Foundation, either version 3 of the License, or     ##
+## (at your option) any later version.                                   ##
+##                                                                       ##
+## This program is distributed in the hope that it will be useful,       ##
+## but WITHOUT ANY WARRANTY; without even the implied warranty of        ##
+## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         ##
+## GNU General Public License for more details.                          ##
+##                                                                       ##
+## You should have received a copy of the GNU General Public License     ##
+## along with this program.  If not, see <http://www.gnu.org/licenses/>. ##
+##                                                                       ##
+###########################################################################
+
+# This analyser is similar to Analyser_Osmosis_Way_Approximate
+# However, this one measures angles and not radiuses
+# It most likely indicates a node that was accidentaly dragged
+# While Analyser_Osmosis_Way_Approximate indicates that the way was drawn
+# with too few nodes
+
+from modules.OsmoseTranslation import T_
+from .Analyser_Osmosis import Analyser_Osmosis
+
+sql10 = """
+WITH
+    points AS (
+        SELECT
+            id,
+            tags->'{tag}' AS type,
+            (ST_DumpPoints(linestring)).geom AS geom4326,
+            (ST_DumpPoints(ST_Transform(linestring, {srid}))).geom AS geom,
+            (ST_DumpPoints(ST_Transform(linestring, {srid}))).path[1] AS index
+        FROM
+            {table_prefix}ways as ways
+        WHERE
+            tags != ''::hstore AND
+            tags?'{tag}' AND tags->'{tag}' IN ('{tag_values}') AND
+            ST_NPoints(linestring) >= 3
+    ),
+    angles AS (
+        SELECT
+            id, index, type,
+            lead(geom4326) OVER (PARTITION BY id ORDER BY index) AS geom4326,
+            lead(geom) OVER (PARTITION BY id ORDER BY index) AS geom,
+            degrees(
+                ST_Angle(geom,
+                     lead(geom, 1) OVER (PARTITION BY id ORDER BY index),
+                     lead(geom, 2) OVER (PARTITION BY id ORDER BY index)
+                )
+            ) angle,
+            ST_Distance(
+                geom,
+                lead(geom, 1) OVER (PARTITION BY id ORDER BY index)
+            ) AS length_1,
+            ST_Distance(
+                lead(geom, 1) OVER (PARTITION BY id ORDER BY index),
+                lead(geom, 2) OVER (PARTITION BY id ORDER BY index)
+            ) AS length_2
+        FROM
+            points
+    )
+
+    SELECT
+        id,
+        ST_AsText(geom4326),
+        angle,
+        type,
+        index
+    FROM
+        angles
+    WHERE
+        (angle < 150 OR angle > 210)
+        AND length_1 < 100
+        AND length_2 < 100
+"""
+
+class Analyser_Osmosis_Rail_Suspicious_Angle(Analyser_Osmosis):
```

Comment:
Just with an eye on the future: 
I guess this analyser (with flexible angle restrictions) could in the future be expanded to include car highways with sharp angles. Maybe better to name it more neutral with `way` instead of `railway`. (Also, I guess it's convenient to have the same file name as analyser class name)

Reply:
I changed it to way, but I not yet convinced that it would make sens for roads and might result in a lot of false positive (e.g. https://www.openstreetmap.org/way/689270247)

Reply:
Yes, for roads (for cars) it should probably be limited to e.g. <30° angles or so, definitely less than 90 or even 150°.

Problem for the future, not for now 😉